internal/validation/validation.go

```diff
@@ -214,6 +215,19 @@ func validateValue(c *opContext, v *common.InputValue, val interface{}, t common
 			}
 		}
 		c.addErr(v.Loc, "VariablesOfCorrectType", "Variable \"%s\" has invalid value %s.\nExpected type \"%s\", found %s.", v.Name.Name, e, t, e)
+	case *schema.InputObject:
+		if val == nil {
+			return
+		}
+		in, ok := val.(map[string]interface{})
+		if !ok {
+			c.addErr(v.Loc, "VariablesOfCorrectType", "Variable \"%s\" has invalid type %T.\nExpected type \"%s\", found %T.", v.Name.Name, val, t, val)
```

Comment:
Why do you print the type (i.e. `%T`) twice? Did you mean to print the value?

Reply:
I've updated this to print the value (quoted). Not sure what I was going for here with reprinting the `%T` value

Reply:
Actually, removed the quoting, since this is not a string value